Addressed issues discussed in team meet - this is ready for final review

samschlicht opened the pull request:
This seems to work fine: linked candidate notes to candidate in server, and then added a method to set a transient field 'intaked' by iterating over the notes. 

I'm not looking to merge just yet, just want to check if this seems like a sensible approach... @camerojo in particular might take a view, since you already had something in mind?

<img width="1182" alt="Screenshot 2023-09-11 at 7 35 36 pm" src="https://github.com/Talent-Catalog/talentcatalog/assets/111261894/eb2c7a20-52cc-411a-9ed7-3b6761e7201c">

samschlicht committed:
This seems to work fine: linked candidate notes to candidate in server, and then added a method to set a transient field 'intaked' by iterating over the notes. Will submit for review to check whether it's a sensible approach!

camerojo reviewed (commented):
Hi @samschlicht - this works well for displaying intaked in results. But the problem is more complicated that that. I believe that they want to have intaked as a search field. That is not possible to do using the Transient field. Also, it gets more complicated for elastic searches.

samschlicht committed:
Addressed issues discussed in team meeting:
- getIntaked() no longer annotated @Nullable (it never returns null)
- getTitle() cannot be null, so there's no need for an escape clause
- introduced frontend functionality to deal with sortable/unsortable fields, thereby preempting the error that clicking to sort a @Transient field would otherwise produce

camerojo reviewed (approved):
Approved

camerojo commented on ui/admin-portal/src/app/model/candidate-field-info.ts:
```diff
@@ -39,11 +39,13 @@ export class CandidateFieldInfo {
-              public fieldSelector: () => boolean) {
+              public fieldSelector: () => boolean,
+              public sortable: boolean) {
     this.displayName = displayName;
```
Well done @samschlicht on making this sortable feature "data driven"
